### PR TITLE
Fix share with guest_app endless loop, add unit tests

### PR DESCRIPTION
**Pull request by micbar**
## Description
Fix endless activity loop when sharing with "guest_app" group and add unit tests.

## Related Issue
https://github.com/owncloud/enterprise/issues/3041

## Motivation and Context
Discussion with @DeepDiver1975 . GroupInterface needs to be implemented correctly.
He requested to add unit tests.

## How Has This Been Tested?
- Manually
- Unit Tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: phpunit.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<phpunit bootstrap="./tests/bootstrap.php"
+		 verbose="true" failOnWarning="true"
+		 timeoutForSmallTests="900"
+		 timeoutForMediumTests="900"
+		 timeoutForLargeTests="900"
+>
+	<testsuite name='guests-unit'>
```
Comment (phil-davis) [inline phpunit.xml]: just call the testsuite ``unit``
The "standard" ``Makefile`` PHP unit command runs the ``unit`` suite

File: tests/bootstrap.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<?php
```
Comment (phil-davis) [inline tests/bootstrap.php]: Can this go in ``tests/unit`` folder, like other apps?
Then it is clearly part of the PHP unit test infrastructure.

**Comment by PVince81** [comment]
@micbar is it intended that this PR brings so much phpunit setup ? the PR description doesn't state so, so it isn't clear whether this was some kind of mistake.

if it was intended, please see @phil-davis's comments.

the fix itself looks fine to me

**Comment by phil-davis** [comment]
@PVince81 Guests currently has no PHP unit tests. So the unfortunate sucker who writes the first PHP unit test gets to also insert the needed CI infrastructure!

**Comment by micbar** [comment]
@DeepDiver1975 requested unit tests, as mentioned in the description.

File: tests/bootstrap.php (hunk 2 above)
Comment (micbar) [reply tests/bootstrap.php]: I can see in the other repos, that the file is in the top folder

**Comment by micbar** [comment]
@phil-davis @PVince81 addressed your input.

**Comment by micbar** [comment]
Currently, we run the unit tests against core master.
I
I assume we would need to run them against stable10 too?

Would be easy to me, to add this. 
Unfortunately, this PR has grown bigger than intended :-)

You decide.

File: tests/bootstrap.php (hunk 2 above)
Comment (micbar) [reply tests/bootstrap.php]: Tom told me in another project, to separate Unit Tests and Integration tests. We would need the bootstrap.php for both test suites?

How is the best practise here?

**Comment by PVince81** [comment]
@micbar if composer.lock changes are not related to your fix, please submit these in a separate PR.

We should keep changes in a PR to the minimum related to the fix, unless really necessary.

**Comment by phil-davis** [comment]
> @micbar if composer.lock changes are not related to your fix, please submit these in a separate PR.
> 
> We should keep changes in a PR to the minimum related to the fix, unless really necessary.

If you do not change ``composer.json`` then there should normally be no change to ``composer.lock`` - I guess you must have done ``composer update`` at some time, rather than just ``composer install``. Unless there is a reason that this PR needs the newer version bumps in ``composer.lock``, then remove it.

File: .drone.yml [hunk 3]
```diff
@@ -91,6 +91,18 @@ pipeline:
...
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    environment:
+      - COVERAGE=${COVERAGE}
+    commands:
+      - if [ -z "${COVERAGE}" ]; then make test-php-unit; fi
+      - if [ "${COVERAGE}" = "true" ]; then make test-php-unit-dbg; fi
+    when:
+      matrix:
+        TEST_SUITE: php-unit
```
Comment (phil-davis) [inline .drone.yml]: Just call it ``phpunit`` to be consistent with what has already been done in all other apps.
That will save people some minor hassle in future when edited drone scripts in multiple repos.

File: .drone.yml [hunk 4]
```diff
@@ -105,6 +117,18 @@ pipeline:
       matrix:
         TEST_SUITE: api-acceptance
 
+  codecov:
+    image: plugins/codecov:2
+    secrets: [codecov_token]
+    pull: true
+    files:
```
Comment (phil-davis) [inline .drone.yml]: Your version will work fine.
But may as well make this section the same as in other apps:
```
  codecov:
    image: plugins/codecov:2
    secrets: [codecov_token]
    pull: true
    paths:
      - tests/output/*.xml
    when:
      matrix:
        COVERAGE: true
```

File: tests/bootstrap.php (hunk 2 above)
Comment (phil-davis) [reply tests/bootstrap.php]: Mostly we do not even need this file. ``phpunit.xml`` uses the one from core directly without change. So try that and see if it works. In other cases I have been putting or moving it into ``tests/unit`` in all the other repos.

File: .drone.yml (hunk 4 above)
Comment (micbar) [reply .drone.yml]: I tried that, didn't work. @patrickjahns recommended to do it this way.

File: .drone.yml (hunk 3 above)
Comment (micbar) [reply .drone.yml]: done

File: tests/bootstrap.php (hunk 2 above)
Comment (micbar) [reply tests/bootstrap.php]: ok, it works with the one from core. Removed it

**Comment by micbar** [comment]
@phil-davis can you review again?

**Comment by phil-davis** [comment]
@PVince81 @DeepDiver1975 are you guys happy with the actual code changes and unit test content?

**Comment by micbar** [comment]
Last remark from my side. Unit tests are testing against OC master branch.

**Pull request merged**